components/eam/src/chemistry/modal_aero/aero_model.F90

```diff
@@ -644,276 +578,7 @@ subroutine aero_model_init( pbuf2d, species_class, iflagaa )
-       num_idx(l) = get_spc_ndx( trim(test_name) )
-       if (num_idx(l) < 0) then
-          write(errmes,fmt='(a,i1)') 'usrrxt_inti: cannot find MAM num_idx ',l
-          write(iulog,*) errmes
-          call endrun(errmes)
-       endif
-    end do
-    dgnumwet_idx = pbuf_get_index('DGNUMWET')
-    if ( aitken_idx < 0 ) then
-       errmes = 'usrrxt_inti: cannot find aitken_idx'
-       call endrun(errmes)
-    end if
-
-    !
-    ! define indeces associated with the various names (defined in
-    ! chemistry/modal_aero/modal_aero_initialize_data.F90)
-    !
-#if ( defined MODAL_AERO_3MODE )
-    !
-    ! accumulation mode #1
-    !
-    index_tot_mass(1,1) = get_spc_ndx('so4_a1')
-    index_tot_mass(1,2) = get_spc_ndx('pom_a1')
-    index_tot_mass(1,3) = get_spc_ndx('soa_a1')
-    index_tot_mass(1,4) = get_spc_ndx('bc_a1' )
-    index_tot_mass(1,5) = get_spc_ndx('dst_a1')
-    index_tot_mass(1,6) = get_spc_ndx('ncl_a1')
-    index_chm_mass(1,1) = get_spc_ndx('so4_a1')
-    index_chm_mass(1,2) = get_spc_ndx('soa_a1')
-    index_chm_mass(1,3) = get_spc_ndx('bc_a1' )
-    !
-    ! aitken mode
-    !
-    index_tot_mass(2,1) = get_spc_ndx('so4_a2')
-    index_tot_mass(2,2) = get_spc_ndx('soa_a2')
-    index_tot_mass(2,3) = get_spc_ndx('ncl_a2')
-    index_chm_mass(2,1) = get_spc_ndx('so4_a2')
-    index_chm_mass(2,2) = get_spc_ndx('soa_a2')
-    !
-    ! coarse mode
-    !
-    index_tot_mass(3,1) = get_spc_ndx('dst_a3')
-    index_tot_mass(3,2) = get_spc_ndx('ncl_a3')
-    index_tot_mass(3,3) = get_spc_ndx('so4_a3')
-    index_chm_mass(3,1) = get_spc_ndx('so4_a3')
-    !
-#elif ( defined MODAL_AERO_4MODE_MOM )
-    !
-    ! accumulation mode #1
-    !
-    index_tot_mass(1,1) = get_spc_ndx('so4_a1')
-    index_tot_mass(1,2) = get_spc_ndx('pom_a1')
-    index_tot_mass(1,3) = get_spc_ndx('soa_a1')
-    index_tot_mass(1,4) = get_spc_ndx('bc_a1' )
-    index_tot_mass(1,5) = get_spc_ndx('dst_a1')
-    index_tot_mass(1,6) = get_spc_ndx('ncl_a1')
-    index_tot_mass(1,7) = get_spc_ndx('mom_a1')
-    index_chm_mass(1,1) = get_spc_ndx('so4_a1')
-    index_chm_mass(1,2) = get_spc_ndx('soa_a1')
-    index_chm_mass(1,3) = get_spc_ndx('bc_a1' )
-    !
-    ! aitken mode
-    !
-    index_tot_mass(2,1) = get_spc_ndx('so4_a2')
-    index_tot_mass(2,2) = get_spc_ndx('soa_a2')
-    index_tot_mass(2,3) = get_spc_ndx('ncl_a2')
-    index_tot_mass(2,4) = get_spc_ndx('mom_a2')
-    index_chm_mass(2,1) = get_spc_ndx('so4_a2')
-    index_chm_mass(2,2) = get_spc_ndx('soa_a2')
-    !
-    ! coarse mode
-    !
-    index_tot_mass(3,1) = get_spc_ndx('dst_a3')
-    index_tot_mass(3,2) = get_spc_ndx('ncl_a3')
-    index_tot_mass(3,3) = get_spc_ndx('so4_a3')
-    index_chm_mass(3,1) = get_spc_ndx('so4_a3')
-    !
-    ! POM mode
-    !
-    index_tot_mass(4,1) = get_spc_ndx('pom_a4')
-    index_tot_mass(4,2) = get_spc_ndx('bc_a4')
-    index_tot_mass(4,3) = get_spc_ndx('mom_a4')
-    index_chm_mass(4,1) = get_spc_ndx('bc_a1' )
-    !
-#elif ( defined MODAL_AERO_4MODE )
```

Comment:
Do we want to get rid of this 4-mode code? Are we not using `index_tot_mass` and `index_chm_mass` anymore?

Reply:
we are not using those indexes so I remove all these lines.